new datastruct workshop (#217)
NuevoFoundation/workshops: content/english/datastruct/Queues/problem 1.md

---
title: "Problem 1: Queue Basics"
date: 2020-02-10T13:24:17-07:00
draft: false
weight: 3
--- 
<link rel="stylesheet" href="../style.css">

> <h1>Task 1: Clearing Queue for New Orders</h1>

Reviewer: ```suggestion
## Task 1: Clearing Queue for New Orders
```

> <p>A new restraunt has been added into Neuvo Eats! New orders have been flooding in and the Queue is flooded with orders! The popularity of the different tacos grew fast and the restraunt is overloaded. Look for a way to clear all elements to save the restraunt!</p>

Reviewer: ```suggestion
<p>A new restaurant has been added into Nuevo Eats! New orders have been flooding in and the Queue is flooded with orders! The popularity of the different tacos grew fast and the restaurant is overloaded. Look for a way to clear all elements to save the restaurant!</p>
```

{{%notice tip%}}
1. How can you iterate through the list?
2. Try to clear the Queue as you go through it!
3. Look at the example menu!
{{%/notice%}}

```js javascript
	
	//this uses a list as the organizer of the queue.
     Queue<String> orders = new LinkedList<>();

     	orders.add("Fish Taco");
     	orders.add("Beef Taco");
     	orders.add("Chicken Taco");
     	orders.add("Fish Taco");
     	orders.add("Beef Taco");

```
<iframe height="800px" width="100%" src="https://replit.com/@nuevofoundation/ClearningQ?lite=true" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>

> <h1>Task 2: Using Priority Queue to find meadian price</h1>

Reviewer: ```suggestion
## Task 2: Using Priority Queue to find median price
```

> <p>A restruant has been using Arrays to hold their list of prices and has a hard time organizing it. They want to switch to a priority queue to make finding the median easier. Turn this array into a priority queue! Then find the median price.</p>

Reviewer: ```suggestion
<p>A restaurant has been using Arrays to hold their list of prices and has a hard time organizing it. They want to switch to a priority queue to make finding the median easier. Turn this array into a priority queue! Then find the median price.</p>
```

---
title: "Problem 1: Queue Basics"
date: 2020-02-10T13:24:17-07:00
draft: false
weight: 3
---

<p style="text-align: center;"><iframe width="60%" height="600px" src="https://www.youtube.com/embed/hITmA5t71N0" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe></p>

<link rel="stylesheet" href="../style.css">

## Task 1: Clearing the Queue

<p>A new restaurant has been added into Nuevo Eats! New orders have been flooding in and the Queue is flooded with orders! The popularity of the different tacos grew fast and the restaurant is overloaded. Look for a way to clear all elements to save the restaurant!</p>

{{%notice tip%}}
1. How can you iterate through the list?
2. Try to clear the Queue as you go through it!
3. What method can help clear a queue?
{{%/notice%}}

```java
	
	   //this uses a list as the organizer of the queue.

>         PriorityQueue<String> orders = new PriorityQueue<String>();

Reviewer: ```suggestion
       PriorityQueue<String> orders = new PriorityQueue<String>();
```

     	orders.add("Fish Taco");
     	orders.add("Beef Taco");
     	orders.add("Chicken Taco");
     	orders.add("Fish Taco");
     	orders.add("Beef Taco");

>         clear(pQueue);

Reviewer: ```suggestion
       clear(pQueue);
```
